src/hotspot/share/nmt/virtualMemoryTracker.hpp

```diff
@@ -296,13 +296,16 @@ class ReservedMemoryRegion : public VirtualMemoryRegion {
 
  public:
   ReservedMemoryRegion(address base, size_t size, const NativeCallStack& stack,
-    MEMFLAGS flag = mtNone) :
+    MEMFLAGS flag) :
     VirtualMemoryRegion(base, size), _stack(stack), _flag(flag) { }
 
 
   ReservedMemoryRegion(address base, size_t size) :
     VirtualMemoryRegion(base, size), _stack(NativeCallStack::empty_stack()), _flag(mtNone) { }
 
+  ReservedMemoryRegion(address base, size_t size, MEMFLAGS flag) :
```

Comment:
(Commenting here because GitHub doesn't allow me to add comments to unchanged lines). I'd like to see a follow-up RFE that makes
```
  ReservedMemoryRegion(address base, size_t size) :
    VirtualMemoryRegion(base, size), _stack(NativeCallStack::empty_stack()), _flag(mtNone) { }
```
a private constructor, so that it is only used for our find operations and never accidentally used for other code.

Reply:
https://bugs.openjdk.org/browse/JDK-8330627 is created for this.